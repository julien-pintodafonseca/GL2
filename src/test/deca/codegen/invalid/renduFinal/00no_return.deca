// Description:
//      Méthode avec un type de retour INT sans instruction return
//      => Toute méthode qui n'est pas de type void doit posséder une instruction return
//
// Resultats:
//      Erreur à l'exécution
//      Ligne 19 : pas d'instruction return

class A {
    float x = 3.5;
    int getX() {
        int y=2;
        this.x=this.x+y;
    }
}

{
    A oA = new A();
    int x = oA.getX();
}
