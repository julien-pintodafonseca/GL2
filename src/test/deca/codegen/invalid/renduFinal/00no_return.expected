Erreur : sortie de la methode A.getX sans instruction return.
